source/extensions/resource_monitors/common/factory_base.h

```diff
@@ -36,6 +36,33 @@ class FactoryBase : public Server::Configuration::ResourceMonitorFactory {
   const std::string name_;
 };
 
+/**
+ * Factory for resource monitors that have empty configuration blocks.
+ */
+class EmptyConfigFactoryBase : public Server::Configuration::ResourceMonitorFactory {
```

Comment:
What's the difference between this and `FactoryBase<Envoy::ProtobufWkt::Empty>`?

Reply:
Envoy::ProtobufWkt::Empty doesn't have a "Validate" method so the call to MessageUtil::downcastAndValidate here (https://github.com/envoyproxy/envoy/blob/master/source/extensions/resource_monitors/common/factory_base.h#L19) doesn't compile.

Reply:
fwiw, there is an http filter config factory that does something similar - https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/common/empty_http_filter_config.h